osquery/tables/system/darwin/system_profiler.mm

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/darwin/cfstring.h>
+#include <osquery/utils/darwin/system_profiler.h>
+#include <osquery/utils/json/json.h>
+
+#include <Foundation/Foundation.h>
+
+namespace osquery {
+namespace tables {
+
+id convertForJSON(id obj) {
```

Comment:
Naive question, but does [`NSJSONSerialization`](https://developer.apple.com/documentation/foundation/jsonserialization?language=objc) make this any easier?

Reply:
It does and it's actually used below but per the docs there is a requirement that 

> All objects are instances of [NSString](https://developer.apple.com/documentation/foundation/nsstring?language=objc), [NSNumber](https://developer.apple.com/documentation/foundation/nsnumber?language=objc), [NSArray](https://developer.apple.com/documentation/foundation/nsarray?language=objc), [NSDictionary](https://developer.apple.com/documentation/foundation/nsdictionary?language=objc), or [NSNull](https://developer.apple.com/documentation/foundation/nsnull?language=objc).

This function converts other object types that were encountered in testing to types that can be serialized by `NSJSONSerialization`.